### DISCUSSION TITLE
Include success or fail dedicated field in workflow telemetry

### DISCUSSION THREAD

**Discussion by CarlOhlsson** [microsoft/AL-Go#475] [Q&A] [answered]
Hi! 

Just set up telemetry last week and did some testing during the weekend. My goal is to find a nice way to monitor CICD/NextMajor/NextMinor workflows for multiple repositories without clicking around in GitHub. I have 15 repositories to keep an eye on and it would be nice to create some kind of dashboard for this. 

Looking at the data in different telemetry signals I was not able to find a dedicated field indicating if the run was successful or not. I can find some workflows that fail with the following kql but that is not a good long-term solution since message contains a lot of other information. Would be nice if we could have a dedicated field to indicate if the workflow was successful or not. 

`traces
| where message contains "failed"
`

Would this be possible, or does it already exist?

**Comment by freddydk** [comment]
I think the telemetry module in GitHub lacks some functionality and is probably a candidate for being refactored.
The above would be a good example of something we should be able to get + test results and bcpt test results etc.

**Comment by rbability** [comment]
Since at least our NextMajor Runs only occur once a week, we set up a subscription for those in a dedicated Teams channel for now. It is not optimal, but at least we have all NextMajor runs in a single place. So on Monday morning you just scroll the list up to see if there is something red in it.
Being able to actually filter somewhere if a run was successful, had warnings or errors would be great of course. But this is what we currently do and it is more comfortable than looking through all repos manually.

**Comment by freddydk** [comment]
@aholstrup1 - do we have these abilities in telemetry today?

**Reply by CarlOhlsson** [reply]
It would be much appreciated if we could find a nice solution to this, currently I’m up to 30+ repos 😄

**Reply by aholstrup1** [reply, marked as answer]
@CarlOhlsson We made some updates to our telemetry story in the last year. Which version of AL-Go are you running? AL-Go versions higher than v6.0 should be using the "new" telemetry which is described here: https://github.com/microsoft/AL-Go/blob/main/Scenarios/EnablingTelemetry.md

With this telemetry you can query for: 
`traces | where message contains "AL-Go workflow failed" ` to get failing workflows. There's also a `WorkflowConclusion` dimension you can filter on.

**Reply by CarlOhlsson** [reply]
Thank you that works, i just created a small KQL query for grouping the events per repository :)
